### DISCUSSION TITLE
questions on how to use mpyc on different host

### DISCUSSION THREAD

**Discussion by shiina3301** [lschoe/mpyc#102] [Q&A] [answered]
I use --config and modify the host name in ini file like party3-0.ini to use mpyc on different host, but it seems to have some error like pid over the limit, what do not appear on local parties using -M.
I want to know how can I run mpyc on different host？

**Comment by lschoe** [comment, marked as answer]
To run MPyC with remote hosts it's maybe easier to use the command line option `-P` to set the hostname and port for each of your parties, in combination with the command line option `-I` to assign a unique index for each party. 

Check out the example for two parties for the `helloworld.py` demo: https://mpyc.readthedocs.io/en/latest/demos.html#helloworld-py

If you want to do this with `ini` files, replacing `localhost` with remote hostnames or IP addresses in the lines `host = localhost` in the `.ini` files should work. 

By leaving `host = `  empty for _exactly one_ of the parties in the `.ini` file, the `pid` will be set to the index of that party. 
Alternatively, you can set `host` for all parties in the `.ini` files and use the command line option `-I i` to set `pid = i`.
